core/domain/collection_services.py

```diff
@@ -936,39 +936,51 @@ def update_collection(
             committer_id, utils.get_current_time_in_millisecs())
 
 
-def create_collection_summary(collection_id, contributor_id_to_add):
+def create_collection_summary(
+        collection_id, contributor_id_to_add, contributor_id_to_remove=None):
     """Creates and stores a summary of the given collection.
 
     Args:
         collection_id: str. ID of the collection.
-        contributor_id_to_add: str. ID of the contributor to be added to the
-            collection summary.
+        contributor_id_to_add: str|None. ID of the contributor to be added to
+            the collection summary.
+        contributor_id_to_remove: str|None. ID of the contributor to remove from
+            the collection summary.
     """
     collection = get_collection_by_id(collection_id)
     collection_summary = compute_summary_of_collection(
-        collection, contributor_id_to_add)
+        collection, contributor_id_to_add, contributor_id_to_remove)
     save_collection_summary(collection_summary)
 
 
-def update_collection_summary(collection_id, contributor_id_to_add):
+def update_collection_summary(
+        collection_id, contributor_id_to_add, contributor_id_to_remove=None):
     """Update the summary of an collection.
 
     Args:
         collection_id: str. ID of the collection.
-        contributor_id_to_add: str. ID of the contributor to be added to the
-            collection summary.
+        contributor_id_to_add: str|None. ID of the contributor to be added to
+            the collection summary.
+        contributor_id_to_remove: str|None. ID of the contributor to remove from
+            the collection summary.
     """
-    create_collection_summary(collection_id, contributor_id_to_add)
+    create_collection_summary(
+        collection_id,
+        contributor_id_to_add,
+        contributor_id_to_remove=contributor_id_to_remove)
 
 
-def compute_summary_of_collection(collection, contributor_id_to_add):
+def compute_summary_of_collection(
+        collection, contributor_id_to_add, contributor_id_to_remove):
     """Create a CollectionSummary domain object for a given Collection domain
     object and return it.
 
     Args:
         collection: Collection. The domain object.
         contributor_id_to_add: str. ID of the contributor to be added to the
             collection summary.
+        contributor_id_to_remove: str|None. ID of the contributor to remove from
```

Comment:
When doing the computation from scratch, better to just ignore users who have been removed? Don't see the need for adding this field.

Reply:
Done.